projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -636,9 +636,11 @@ export class IgxGridSelectionService {
             owner: this.grid,
             oldSelection: currSelection,
             newSelection,
-            added, removed,
-            event, cancel: false,
-            allRowsSelected: this.areAllRowSelected(newSelection.map(r =>  this.getRecordKey(r)))
+            added,
+            removed,
+            event,
+            cancel: false,
+            allRowsSelected: this.areAllRowSelected(newSelection)
```

Comment:
With this approach, the test `Should be able to programmatically overwrite the selection using rowSelectionChanging event` failed because `newSelection` was mapped using `getRecordKey` before emitting the event. This caused a format mismatch between the original selection (`oldSelection`) and the modified selection (`newSelection`). As a result, the test logic that tries to prevent even `ProductID` rows from being selected (`e.newSelection = e.oldSelection || []`) did not work correctly. The test couldn't properly overwrite the selection, leading to the failure.